app/services/idv/actions/verify_document_action.rb

```diff
@@ -33,19 +33,31 @@ def enqueue_job
           verify_document_capture_session_uuid_key,
         )
         document_capture_session.requested_at = Time.zone.now
-        document_capture_session.store_proofing_pii_from_doc({})
+        document_capture_session.store_proofing_pii_from_doc({}) # generates a result_id
 
-        VendorDocumentVerificationJob.perform(
-          document_capture_session_result_id: document_capture_session.result_id,
-          encryption_key: params[:encryption_key],
-          front_image_iv: params[:front_image_iv],
-          back_image_iv: params[:back_image_iv],
-          selfie_image_iv: params[:selfie_image_iv],
-          front_image_url: params[:front_image_url],
-          back_image_url: params[:back_image_url],
-          selfie_image_url: params[:selfie_image_url],
-          liveness_checking_enabled: params[:liveness_checking_enabled],
+        callback_url = Rails.application.routes.url_helpers.document_proof_result_url(
+          result_id: document_capture_session.result_id,
         )
+
+        LambdaJobs::Runner.new(
+          job_class: Idv::Proofer.document_job_class,
+          args: {
+            encryption_key: params[:encryption_key],
+            front_image_iv: params[:front_image_iv],
+            back_image_iv: params[:back_image_iv],
+            selfie_image_iv: params[:selfie_image_iv],
+            front_image_url: params[:front_image_url],
+            back_image_url: params[:back_image_url],
+            selfie_image_url: params[:selfie_image_url],
+            liveness_checking_enabled: params[:liveness_checking_enabled],
```

Comment:
One thing I noticed in trying to test this in the browser is that we don't actually send an explicit parameter for this from the client. We _could_, or it also seems we could also reasonably derive this from the presence of a `selfie_image_url`. If we did that, not sure if it makes sense to consider here, or from within `identity-idp-functions`. I'd lean toward `identity-idp-functions` to have fewer things to wrangle.

Reply:
Great catch!

Since this this derives from `DocAuthBaseStep` we can just derive this from the server:

https://github.com/18F/identity-idp/blob/97fe583ff97ba1a8e6c43b6bafda160b9b9cc6e4/app/services/idv/steps/doc_auth_base_step.rb#L174-L176

so even if the client sends a URL when it shouldn't, it wouldn't change the behavior?

Reply:
Fixed in a62f45097ee9a9267bb0666a8470ab1aa50c537f